ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -38,22 +37,19 @@ bool joystick_exists(int id) {
 }
 
 std::string joystick_name(int id) {
-  const char *name;
-  name = SDL_JoystickName(joystick_get_handle(id));
+  const char *name = SDL_JoystickName(joystick_get_handle(id));
   return (name ? name : "");
```

Comment:
nit: ```name == nullptr``` for clarity

Reply:
Do we want to print any DEBUG_MESSAGE's with the contents of SDL_GetError()? This would a nice application to use that, in case the joystick name can't be retrieved for whatever reason, it would print more specifically what the exact problem was that we might not know without peeking at SDL's source and making our own errors, which would be a waste of time imo, since errors are already provided by the api.

https://wiki.libsdl.org/SDL_GetError

Reply:
if you fail to get the name you can print a debug message here too if you wish. SDL_GetErrror() just returns const char* so you should be able to use it directly in DEBUG_MESSAGE.  However, you will need a std::string if you want to concat it.
ie ```DEBUG_MESSAGE(std::string("Error: enigma took a shit ")  + SDL_GetError());```